src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -831,12 +831,34 @@ int muxer_usage(bool is_sdk_present)
 }
 
 // Convert "path" to realpath (merging working dir if needed) and append to "realpaths" out param.
-void append_realpath(const pal::string_t& path, std::vector<pal::string_t>* realpaths)
+void append_probe_realpath(const pal::string_t& path, std::vector<pal::string_t>* realpaths, const pal::string_t& tfm)
 {
-    pal::string_t real = path;
-    if (pal::realpath(&real))
+    pal::string_t probe_path = path;
+
+    if (pal::realpath(&probe_path) && pal::directory_exists(probe_path))
```

Comment:
Can I not create a folder that as "{arch}/{tfm}" in its name such that the call to directory exists succeeds? I think you need to search for the pattern first, transform it if applicable and then use the resultant value for the folder check.